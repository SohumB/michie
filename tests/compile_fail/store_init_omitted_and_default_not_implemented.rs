use core::marker::PhantomData;
use michie::{memoized, MemoizationStore};

struct Store<K, V> {
    k: PhantomData<K>,
    v: PhantomData<V>,
}
impl<K, V> MemoizationStore<K, V> for Store<K, V> {
    fn insert(&mut self, _key: K, _value: V) {}
    fn get(&self, _key: &K) -> Option<&V> {
        None
    }
}
#[memoized(key_expr = input, store_type = Store)]
fn f(input: usize) -> usize {
    input
}

fn main() {}
