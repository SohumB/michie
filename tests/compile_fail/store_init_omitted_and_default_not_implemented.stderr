error[E0277]: the trait bound `Store<usize, usize>: Default` is not satisfied
  --> tests/compile_fail/store_init_omitted_and_default_not_implemented.rs:14:1
   |
14 | #[memoized(key_expr = input, store_type = Store)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `Store<usize, usize>`
   |
   = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)
